### PR TITLE
[ML] Remove Version from MlConfigVersion completely

### PR DESCRIPTION
**Pull request by droberts195**
Proof-of-concept to show how Version can be removed from MlConfigVersion if DiscoveryNode is altered to make available the legacy version ID for 8.10.0 or older nodes. DiscoveryNode will have to read this integer off the wire when serialized from these old nodes, so even if Version is removed from DiscoveryNode it will be able to provide the required value.

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/MlConfigVersion.java [hunk 1]
```diff
@@ -329,12 +318,17 @@ public static MlConfigVersion fromString(String str) {
         if (str.startsWith("8.10.") || str.equals("8.11.0")) {
             return V_10;
         }
-        Matcher matcher = Pattern.compile("^(\\d+)\\.0\\.0$").matcher(str);
-        int versionNum;
-        if (matcher.matches() == false || (versionNum = Integer.parseInt(matcher.group(1))) < 10) {
-            return fromVersion(Version.fromString(str));
+        Matcher matcher = Pattern.compile("^(\\d+)\\.(\\d+)\\.(\\d+)(?:-\\w+)?$").matcher(str);
```
Comment (rjernst) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/MlConfigVersion.java:321]: I might have missed this from an earlier review, but why does ML config version look like a semantic version? Could it just be an integer going forward, like other version concepts we have now (transport version, index version, etc)?

Comment (droberts195) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/MlConfigVersion.java:321]: Old nodes in mixed version clusters need to be able to parse the text representation of the newer versions without throwing exceptions.

**Comment by droberts195** [comment]
@elasticmachine update branch

File: server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java [hunk 2]
```diff
@@ -488,6 +488,16 @@ public Version getVersion() {
         return this.versionInfo.nodeVersion();
     }
 
+    public int getLegacyVersionIdOrThrow() {
```
Comment (thecoop) [inline server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java:491]: Could return `OptionalInt` here, leaving it up to the caller to determine what should be done if its out of bounds

**Review by rjernst — APPROVED** [review]
This looks like a pragmatic approach to migrating from legacy versions. LGTM.

**Pull request merged**